main.go

```diff
@@ -29,139 +30,221 @@ type backup struct {
 	in_use                 bool
 }
 
-func main() {
-
-	//get configuration options from environment variables
-	dailyBackupsRetained, err := strconv.Atoi(os.Getenv("DAILY_BACKUPS"))
-	if err != nil {
-		log.Fatalf("Error retrieving DAILY_BACKUPS env var: %v", err)
+type backupconfig struct {
+	Name              string `yaml:"name"`
+	KanisterNamespace string `yaml:"kanisterNamespace"`
+	BlueprintName     string `yaml:"blueprintName"`
+	ProfileName       string `yaml:"profileName"`
+	Retention         struct {
+		Backups string `yaml:"backups"`
+		Minutes string `yaml:"minutes"`
+		Hours   string `yaml:"hours"`
+		Days    string `yaml:"days"`
+		Months  string `yaml:"months"`
+		Years   string `yaml:"years"`
 	}
+}
 
-	weeklyBackupsRetained, err := strconv.Atoi(os.Getenv("WEEKLY_BACKUPS"))
+func main() {
+	// creates the in-cluster config
+	config, err := rest.InClusterConfig()
 	if err != nil {
-		log.Fatalf("Error retrieving WEEKLY_BACKUPS env var: %v", err)
-	}
-
-	kanisterNamespace := os.Getenv("KANISTER_NAMESPACE")
-	if len(kanisterNamespace) == 0 {
-		log.Fatalf("KANISTER_NAMESPACE value cannot be empty")
-	}
-
-	blueprintName := os.Getenv("BLUEPRINT_NAME")
-	if len(blueprintName) == 0 {
-		log.Fatalf("BLUEPRINT_NAME value cannot be empty")
-	}
-
-	s3ProfileName := os.Getenv("S3_PROFILE_NAME")
-	if len(s3ProfileName) == 0 {
-		log.Fatalf("S3_PROFILE_NAME value cannot be empty")
+		panic(err.Error())
 	}
 
-	//set env schedule
-	const evalSchedule string = "1/5 * * * *"
-
-	//creates the in-cluster config
-	config, err := rest.InClusterConfig()
+	// creates the out-of-cluster config
+	// var kubeconfig *string
+	// if home := homedir.HomeDir(); home != "" {
+	// 	kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+	// } else {
+	// 	kubeconfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	// }
+	// flag.Parse()
+
+	// use the current context in kubeconfig
+	// config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	// if err != nil {
+	// 	panic(err.Error())
+	// }
+
+	// initialise dynamicclient
+	dynamicClient, err := dynamic.NewForConfig(config)
 	if err != nil {
 		panic(err.Error())
 	}
 
-	//initialise dynamicclient
-	dynamicClient, err := dynamic.NewForConfig(config)
+	// create the clientset
+	clientset, err := kubernetes.NewForConfig(config)
 	if err != nil {
 		panic(err.Error())
 	}
 
-	//specify the crds which should be queried
+	// specify the crds which should be queried
 	gvr := schema.GroupVersionResource{
 		Group:    "cr.kanister.io",
 		Version:  "v1alpha1",
 		Resource: "actionsets",
 	}
 
-	log.Printf("Config options\nRetaining %v daily, %v weekly backups\n", dailyBackupsRetained, weeklyBackupsRetained)
-	log.Printf("Kanister namespace: %v\n", kanisterNamespace)
-	log.Printf("Blueprint name: %v\n", blueprintName)
-	log.Printf("S3 profile name: %v\n", s3ProfileName)
+	// prometheus gauge vector definition
```

Comment:
I suggest to chuck the metrics initialization in a function and also have those gauge vectors be grouped up in a struct (e.g. `metrics` or `taweretMetrics`) so that we can pass it around less verbosely and we can simply expand it in the future if needed.

Reply:
done!